### PR TITLE
fix(bootstrap): KMS keys cannot be tagged

### LINKED ISSUE #21281: CLI: Bootstrapping fails when adding tags to existing KMS key
### Describe the bug

When re-bootstrapping an existing CDKToolkit stack that contains a customer managed KMS key, the update fails if tags are added.

### Expected Behavior

The stack update should be successful and the tags should be applied to the existing resources.

### Current Behavior

The stack update fails because the file bucket encryption key lacks the `kms:TagResource` permission.

### Reproduction Steps

* Bootstrap without tags:
  * `cdk bootstrap --bootstrap-customer-key  aws://<account-id>/<region>`
* Update with tags:
  * `cdk bootstrap --bootstrap-customer-key  aws://<account-id>/<region> --tags foo=bar`
  * Update fails with the error `"Access denied for operation 'TagResource'."`

### Possible Solution

Add the permissions for KMS tagging/untagging to the customer key.

### Additional Information/Context

_No response_

### CDK CLI Version

2.33.0

### Framework Version

_No response_

### Node.js Version

16.x

### OS

MacOS 12.4

### Language

Typescript

### Language Version

_No response_

### Other information

_No response_

### PR DESCRIPTION
**Pull request by adrianmace**
This change allows IAM users and roles within the account to update and replace the tags on the provided KMS key.

If the bootstrap command is run and custom tags are specified, these are added to the KMS key. If the command is run additional times with different tags, the key policy needs to allow the replacing / updating of these tags so that it succeeds.

Fixes #21281 

**Note for reviewer:** Due to performance limitations of my current device, I have been unable to test these changes. I welcome edits by maintainers as necessary.

----

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md)

### Adding new Unconventional Dependencies:

* [ ] This PR adds new unconventional dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md/#adding-new-unconventional-dependencies)

### New Features

* [x] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md)?
	* [ ] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Review by TheRealAmazonKendra — DISMISSED** [review]
Thank you for your contribution. Unfortunately, we cannot accept untested changes and do not have the capacity to work on p2 issues ourselves. If any other contributors would like to pitch in here to help test this, we can provide a review at that time.

File: packages/aws-cdk/lib/api/bootstrap/bootstrap-template.yaml [hunk 1]
```diff
@@ -106,6 +106,8 @@ Resources:
               - kms:ScheduleKeyDeletion
               - kms:CancelKeyDeletion
               - kms:GenerateDataKey
+              - kms:TagResource
+              - kms:UntagResource
```
Comment (Naumel) [inline packages/aws-cdk/lib/api/bootstrap/bootstrap-template.yaml:110]: Hello,

From the issue I can understand why the [`kms:TagResource`](https://docs.aws.amazon.com/cli/latest/reference/kms/tag-resource.html) is required.

>the update fails if tags are added.

`"Access denied for operation 'TagResource'."`


Mind providing the use case that requires untagging/updating?

Cheers!

Comment (adrianmace) [reply packages/aws-cdk/lib/api/bootstrap/bootstrap-template.yaml:110]: #21571 goes into detail about the use case where a tag changes.

Though I admit I made an assumption that the uderlying Cloudformation would attempt an `UntagResource` action at some point during the updating process.

**Comment by Naumel** [comment]
@mergifyio

**Comment by adrianmace** [comment]
Just taking a look at this on my phone while AFK. 

I’ve just noticed that it is possible that the test I’ve included may not work, and may require `force: true` on the second bootstrap invocation.

**Comment by adrianmace** [comment]
@Mergifyio update

**Comment by adrianmace** [comment]
@rix0rrr thank you for merging.

Out of curiosity, is there anything that needs to be done to dictate a new bootstrap template _version_ or is that handled by the release process?

**Comment by adrianmace** [comment]
@rix0rrr I get the feeling I should have also bumped this value here.

https://github.com/aws/aws-cdk/blob/0e552dbb63a97cd6a7a65cae80ae863609237e61/packages/aws-cdk/lib/api/bootstrap/bootstrap-template.yaml#L590

**Comment by joelcox22** [comment]
I have had similar questions in another PR (that didn't end up getting merged, which is OK) - https://github.com/aws/aws-cdk/pull/23004#issuecomment-1321327231

If there is clarity on what the expectations are for bootstrap stack updates, I'd love to see it documented in the contributing guide or something.

**Pull request merged**